src/utils/storage.js

```diff
@@ -0,0 +1,39 @@
+const storageCart = {
+  get(product) {
+    try {
+      const rawValue = window.localStorage.getItem(product);
+      return JSON.parse(rawValue);
+    } catch (error) {
+      console.error(`Error parsing storage item "${product}".`);
+      return null;
+    }
+  },
+
+  set(value) {
+    const rawValue = JSON.parse(window.localStorage.getItem("products"));
+    const newVal = JSON.parse(JSON.stringify(value));
+    if (rawValue === null) {
+      var jsons = [];
+      jsons.push(newVal);
+      window.localStorage.setItem("products", JSON.stringify(jsons));
+    } else {
+      var jsonsVal = rawValue;
+      jsonsVal.push(newVal);
```

Comment:
Here we can check if we already have this product ID in the cart, and if we do, we just add the quantity to the current stockOnCartValue. This way we prevent having the same item several times on the Cart and Checkout page.